assets/js/hooks/media_player.js

```diff
@@ -18,16 +18,21 @@ MediaPlayer = {
     this.duration = this.el.querySelector("#player-duration")
     this.progress = this.el.querySelector("#player-progress")
 
-    this.handleEvent("update_display_value", ([key, val, extraKey]) => {
-      if (extraKey === "innerText") {
-        this[key][extraKey] = val;
-      }
+    this.el.addEventListener("update_display_value", e => this.handleUpdateDisplayValue(e))
+  },
+  handleUpdateDisplayValue(e) {
+    const {
+      detail,
+    } = e
+    const [key, val, extraKey] = detail?.payload
```

Comment:
todo: add guard instead of doing a tuple de-structure here because if detail?.payload is null or undefined, the iterator used for destructuring will throw a type exception